pkg/state/state_skel.go

```diff
@@ -91,32 +92,103 @@ func (s *stateSkel) updateObj(obj *unstructured.Unstructured) error {
 	return nil
 }
 
+// When updating, not all resources need an updated ResourceVersion
+func needToUpdateResourceVersion(kind string) bool {
+	if kind == "SecurityContextConstraints" ||
+		kind == "Service" ||
+		kind == "ServiceMonitor" ||
+		kind == "Route" ||
+		kind == "BuildConfig" ||
+		kind == "ImageStream" ||
+		kind == "PrometheusRule" {
+		return true
+	}
+	return false
+}
+
+func checkNestedFields(found bool, err error) {
+	if !found || err != nil {
+		log.Error(err, "")
+		os.Exit(1)
+	}
+}
+
+func updateResourceVersion(req, found *unstructured.Unstructured) error {
+	kind := found.GetKind()
+
+	if needToUpdateResourceVersion(kind) {
+		version, fnd, err := unstructured.NestedString(found.Object, "metadata", "resourceVersion")
+		checkNestedFields(fnd, err)
+
+		if err := unstructured.SetNestedField(req.Object, version, "metadata", "resourceVersion"); err != nil {
+			return errors.Wrap(err, "Couldn't update ResourceVersion")
+		}
+	}
+
+	if kind == "Service" {
```

Comment:
can you add a comment on why we need to update ClusterIP ?

also i see you are just updating the field with the same value, is that ok ?

im probably lacking some k8s specific knowledge about `ClusterIP ` so apologies if the questions seem obvious

Reply:
Just lessons learned from other operator issues, I will check into why this is required for `Services`.